server.py

```diff
@@ -360,6 +366,49 @@ def healthcheck():
         return jsonify({'status': 'OK'})
 
 
+def get_all_comments_by_survey_id(survey_id):
+    # collect survey based on survey id and and 146 code
+
+    db.engine.echo = True
+    max_surveys = 10
+
+    # retrieve all the records
+    comments = list()
+    sliced_value = 0
+    while True:
+        survey_records_page = db.session.query(SurveyResponse).slice(sliced_value, max_surveys).all()
+        if not survey_records_page or len(survey_records_page) == 0:
+            break
+
+        for data in survey_records_page:
+            logger.debug(data.data)
+            record = json.loads(data.data)
+            if '146' in record['data']:
+                if survey_id in record['survey_id']:
+                    comments.append((record['survey_id'], record['data']['146']))
+        sliced_value = len(survey_records_page)
+    logger.info("Comments retrieved " + str(len(comments)))
+    return comments
+
+
+@app.route('/comments/<string:survey_id>', methods=['GET'])
+def get_comments(survey_id):
+    logger.info("Exporting comments for survey id " + survey_id)
+
+    if survey_id is None:
```

Comment:
if survey_id is passed as an empty string, wont we want this to error also? which this wont?

Reply:
So eq used allow empty string `Survey ID's` but have put a rule in validator where its enforced that there of a certain form, *not* an empty string, so we should be ok the way angelo has done it